sdk/communication/communication-phone-numbers/test/utils/matrix.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * @internal
+ * Takes a jagged 2D array and a function and runs the function with every
+ * possible combination of elements of each of the arrays
+ *
+ * For strong type-checking, it is important that the `matrix` have a strong
+ * type, such as a `const` literal.
+ *
+ * @param values - jagged 2D array specifying the arguments and their possible
+ *                 values
+ * @param handler - the function to run with the different argument combinations
+ *
+ * @example
+ * ```typescript
+ * matrix([
+ *     [true, false],
+ *     [1, 2, 3]
+ *   ] as const,
+ *   (useLabels: boolean, attempts: number) => {
+ *     // This body will run six times with the following parameters:
+ *     // - true, 1
+ *     // - true, 2
+ *     // - true, 3
+ *     // - false, 1
+ *     // - false, 2
+ *     // - false, 3
+ *   });
+ * ```
+ */
+export function matrix<T extends ReadonlyArray<readonly unknown[]>>(
```

Comment:
@deyaaeldeen @willmtemple 
We're stealing this nifty code for a couple of our services, is there a common place for the entire SDK where it could be shared?

Reply:
Perhaps it could live here? https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/test-utils/multi-version/ ? 

Or we could make a new test-utils package